invenio_rdm_records/contrib/codemeta/custom_fields.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+"""CodeMeta specific custom fields.
+
+Implements the following fields:
+- Repository URL
+- Programming language
+- Runtime platform
+- Operating system
+- Development status
+"""
+
+from flask_babelex import _
+from invenio_records_resources.services.custom_fields import KeywordCF, TextCF
+from invenio_records_resources.services.records.facets import CFTermsFacet
+from invenio_vocabularies.services.custom_fields import VocabularyCF
+from invenio_vocabularies.services.facets import VocabularyLabels
+from marshmallow import validate
+
+CODE_NAMESPACE = {
+    # CodeMeta
+    "code": "https://codemeta.github.io",
+}
+
+
+CODE_CUSTOM_FIELDS = [
+    TextCF(
+        name="code:codeRepository",
+        field_args={
+            "validate": validate.URL(),
+            "error_messages": {"validate": "You must provide a valid URL."},
+        },
+    ),
+    KeywordCF(name="code:programmingLanguage"),
+    TextCF(name="code:runtimePlatform"),
+    TextCF(name="code:operatingSystem"),
+    VocabularyCF(
+        name="code:developmentStatus",
+        vocabulary_id="developmentstatus",
+        dump_options=True,
+        multiple=False,
+    ),
+]
+
+CODE_CUSTOM_FIELDS_UI = {
+    "section": _("CodeMeta"),
+    "fields": [
+        dict(
+            field="code:codeRepository",
+            ui_widget="Input",
+            props=dict(
+                label="Repository URL",
+                icon="linkify",
+                description="You should fill this field with the repository URL.",
```

Comment:
Fields in the RDM deposit form have a explanatory nature, with no verb or "order" to the user. For consistency I would change descriptions to something like:
```suggestion
                description="URL or link where the code repository is hosted.",
```

Now the challenge comes from writing a sentence to define a field without repeating the name of the field 😂 Feel free to change the provided suggestion.